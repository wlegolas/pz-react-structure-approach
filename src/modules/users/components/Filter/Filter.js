import PropTypes from 'prop-types'
import React from 'react'

const Filter = ({ onKeyUp }) => {
  return (
    <div className="pz-search">
      <input
        type="text"
        placeholder="Fill the github user and press enter to search"
        onKeyUp={onKeyUp}
      />
    </div>
  )
}

Filter.propTypes = {
  onKeyUp: PropTypes.func.isRequired
}

export default Filter
