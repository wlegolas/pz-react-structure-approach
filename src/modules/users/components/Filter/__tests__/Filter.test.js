import toJson from 'enzyme-to-json'
import React from 'react'
import Filter from '../Filter'

describe('<Filter />', () => {
  let wrapper

  beforeAll(() => {
    const handleKeyUp = jest.fn()
    wrapper = shallow(<Filter onKeyUp={handleKeyUp} />)
  })

  describe('render()', () => {
    it('should render the component', () => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('verify the structure', () => {
    it('should have a container with the class "pz-search"', () => {
      expect(wrapper.find('div.pz-search')).toHaveLength(1)
    })

    it('should have an input element', () => {
      expect(wrapper.find('input')).toHaveLength(1)
    })

    it('should have an input with the attribute "placeholder" defined', () => {
      expect(wrapper.find('input').prop('placeholder')).toBe(
        'Fill the github user and press enter to search'
      )
    })
  })

  describe('onKeyUp', () => {
    it('should call the function when event "keyup" is fired', () => {
      const handleKeyUp = jest.fn()
      wrapper = shallow(<Filter onKeyUp={handleKeyUp} />)

      wrapper.find('input').simulate('keyup')

      expect(handleKeyUp).toHaveBeenCalledTimes(1)
    })

    it('should not call the functions when event "keydown" is fired', () => {
      const handleKeDown = jest.fn()
      wrapper = shallow(<Filter onKeyUp={handleKeDown} />)

      wrapper.find('input').simulate('keydwon')

      expect(handleKeDown).toHaveBeenCalledTimes(0)
    })
  })
})
