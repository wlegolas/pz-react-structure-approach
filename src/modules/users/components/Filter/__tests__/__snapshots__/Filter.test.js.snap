// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> render() should render the component 1`] = `
<div
  className="pz-search"
>
  <input
    onKeyUp={[MockFunction]}
    placeholder="Fill the github user and press enter to search"
    type="text"
  />
</div>
`;
